### PR TITLE
chore: Fix sonar issues

### PR DESCRIPTION
**Pull request by david-mackessy**

### CODE REVIEW

File: dhis-2/dhis-support/dhis-support-artemis/src/main/java/org/hisp/dhis/artemis/config/NameDestinationResolver.java [hunk 1]
```diff
@@ -52,22 +52,10 @@ public Destination resolveDestinationName(
...
 
   private Destination resolveTopic(Session session, String topicName) throws JMSException {
-    if (session instanceof TopicSession) {
-      // Cast to TopicSession: will work on both JMS 1.1 and 1.0.2
-      return session.createTopic(topicName);
-    } else {
-      // Fall back to generic JMS Session: will only work on JMS 1.1
-      return session.createTopic(topicName);
-    }
+    return session.createTopic(topicName);
```
Comment (david-mackessy) [inline dhis-2/dhis-support/dhis-support-artemis/src/main/java/org/hisp/dhis/artemis/config/NameDestinationResolver.java:55]: both branches do the same thing so collapsing to use once

File: dhis-2/dhis-support/dhis-support-artemis/src/main/java/org/hisp/dhis/artemis/config/NameDestinationResolver.java [hunk 2]
```diff
@@ -52,22 +52,10 @@ public Destination resolveDestinationName(
...
 
   private Queue resolveQueue(Session session, String queueName) throws JMSException {
-    if (session instanceof QueueSession) {
-      // Cast to QueueSession: will work on both JMS 1.1 and 1.0.2
-      return session.createQueue(queueName);
-    } else {
-      // Fall back to generic JMS Session: will only work on JMS 1.1
-      return session.createQueue(queueName);
-    }
+    return session.createQueue(queueName);
```
Comment (david-mackessy) [inline dhis-2/dhis-support/dhis-support-artemis/src/main/java/org/hisp/dhis/artemis/config/NameDestinationResolver.java:59]: both branches do the same thing so collapsing to use once

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/SystemSettingController.java [hunk 3]
```diff
@@ -109,15 +108,13 @@ public WebMessage setSystemSettingOrTranslation(
       @PathVariable(value = "key") String key,
       @RequestParam(value = "locale", required = false) String locale,
       @RequestParam(value = "value", required = false) String value,
-      @RequestBody(required = false) String valuePayload,
-      HttpServletResponse response,
-      HttpServletRequest request)
```
Comment (david-mackessy) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/SystemSettingController.java:114]: neither params are used

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/SystemSettingController.java [hunk 4]
```diff
@@ -235,6 +232,8 @@ private boolean settingExistsAndIsNotConfidential(String key) {
   private Serializable getSystemSettingOrTranslation(String key, String locale, User currentUser) {
     if (settingExistsAndIsNotConfidential(key)) {
       Optional<SettingKey> settingKey = SettingKey.getByName(key);
+      if (settingKey.isEmpty()) return StringUtils.EMPTY;
```
Comment (david-mackessy) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/SystemSettingController.java:235]: returning an empty string is the pattern applied in this method when something not found so keeping it consistent here

**Review by netroms — APPROVED** [review]

**Review by jbee — APPROVED** [review]

**Pull request merged**